x/mint/keeper/keeper_test.go

```diff
@@ -60,109 +60,131 @@ func TestKeeperTestSuite(t *testing.T) {
 }
 
 func (suite *KeeperTestSuite) TestMintCoinsToFeeCollectorAndGetProportions() {
-	mintKeeper := suite.App.MintKeeper
-
-	// When coin is minted to the fee collector
-	fee := sdk.NewCoin("stake", sdk.NewInt(0))
-	fees := sdk.NewCoins(fee)
-	coin := mintKeeper.GetProportions(suite.Ctx, fee, sdk.NewDecWithPrec(2, 1))
-	suite.Equal("0stake", coin.String())
-
-	// When mint the 100K stake coin to the fee collector
-	fee = sdk.NewCoin("stake", sdk.NewInt(100000))
-	fees = sdk.NewCoins(fee)
-
-	err := simapp.FundModuleAccount(suite.App.BankKeeper,
-		suite.Ctx,
-		authtypes.FeeCollectorName,
-		fees)
-	suite.NoError(err)
-
-	// check proportion for 20%
-	coin = mintKeeper.GetProportions(suite.Ctx, fee, sdk.NewDecWithPrec(2, 1))
-	suite.Equal(fees[0].Amount.Quo(sdk.NewInt(5)), coin.Amount)
-}
-
-func (suite *KeeperTestSuite) TestDistrAssetToDeveloperRewardsAddrWhenNotEmpty() {
-	mintKeeper := suite.App.MintKeeper
-	params := suite.App.MintKeeper.GetParams(suite.Ctx)
-	devRewardsReceiver := sdk.AccAddress([]byte("addr1---------------"))
-	gaugeCreator := sdk.AccAddress([]byte("addr2---------------"))
-	devRewardsReceiver2 := sdk.AccAddress([]byte("addr3---------------"))
-	devRewardsReceiver3 := sdk.AccAddress([]byte("addr4---------------"))
-	params.WeightedDeveloperRewardsReceivers = []types.WeightedAddress{
+	tests := []struct {
+		name                 string
+		ratio                sdk.Dec
+		hasPreExistingSupply bool
+		expectedCoin         sdk.Coin
+		fee                  sdk.Coin
+	}{
 		{
-			Address: devRewardsReceiver.String(),
-			Weight:  sdk.NewDec(1),
+			name:                 "coin is minted to the fee collector",
+			fee:                  sdk.NewCoin("stake", sdk.NewInt(0)),
+			ratio:                sdk.NewDecWithPrec(2, 1),
```

Comment:
consider adding test cases for testing different ratios || removing this field and using default constants!